tensorflow_datasets/scripts/replace_fake_images.py

```diff
@@ -104,20 +101,24 @@ def rewrite_tar(root_dir, tar_filepath):
     # Checking the extension of file to be extract
     if tar_filepath.lower().endswith('gz'):
       extension = ':gz'
+      shutil_extension = 'gztar'
+      extension_removal_index = 7 # Hardcode number of indices to be skipped from the end to obtain filename. This is hard-coded as filenames containing "." might exist and thus filepath.split(".") would give an incorrect result.
```

Comment:
This won't works with `.tgz` files. Even if we may not have now, we may in the future. Is there a way to specify the extension in `make_archive` ?

Reply:
Unfortunately, there is no way to specify the target extension to make_archive.
There are 5 options: 
1. 'zip' - .zip extension
2.'tar' - .tar extension
3.'gztar' - .tar.gz extension
4.'bztar' - .tar.bz2 extension
5.'xztar' - .tar.xz extension

As you already know .tgz is equivalent to .tar.gz. So will renaming a file from .tgz to tar.gz be a major issue in the future?

Reply:
> As you already know .tgz is equivalent to .tar.gz. So will renaming a file from .tgz to tar.gz be a major issue in the future?

It would be. We don't control the dataset filenames. For example, if imagenet use `train.tgz`, then the fake data should match this name too.

Reply:
I have tested this with the latest commit, .tgz files will retain their extension without changing to .tar.gz. The same is true for other tar compression variants as well. So marking this issue as resolved.